### ISSUE TITLE
podman machine not starting on macOS Catalina 10.15.7 (amd64)

### ISSUE THREAD

**Issue by cfelder** [containers/podman#16048]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

First of all, thanks for creating macOS Pkg files as a brew alternative. On a recent macOS Monterey 12.3.1 (amd64) this is working fine. On my macOS Catalina 10.15.7 (amd64) the following fails though.

**Steps to reproduce the issue:**

1. Install [podman-installer-macos-amd64.pkg v4.2.1](https://github.com/containers/podman/releases/download/v4.2.1/podman-installer-macos-amd64.pkg)

2. ``/opt/podman/bin/podman machine init``

3. ``/opt/podman/bin/podman machine start``

**Describe the results you received:**

```
$ /opt/podman/bin/podman --log-level debug machine start
INFO[0000] /opt/podman/bin/podman filtering at log level debug
Starting machine "podman-machine-default"
[/opt/podman/qemu/bin/gvproxy -listen-qemu unix:///var/folders/mf/fk8hldv12_b3_j1vnp464th80000gn/T/podman/qmp_podman-machine-default.sock -pid-file /var/folders/mf/fk8hldv12_b3_j1vnp464th80000gn/T/podman/podman-machine-default_proxy.pid -ssh-port 61062 -forward-sock /Users/felder/.local/share/containers/podman/machine/podman-machine-default/podman.sock -forward-dest /run/user/501/podman/podman.sock -forward-user core -forward-identity /Users/felder/.ssh/podman-machine-default --debug]
DEBU[0000] qemu cmd: [/opt/podman/qemu/bin/qemu-system-x86_64 -m 2048 -smp 1 -fw_cfg name=opt/com.coreos/config,file=/Users/felder/.config/containers/podman/machine/qemu/podman-machine-default.ign -qmp unix:/var/folders/mf/fk8hldv12_b3_j1vnp464th80000gn/T/podman/qmp_podman-machine-default.sock,server=on,wait=off -netdev socket,id=vlan,fd=3 -device virtio-net-pci,netdev=vlan,mac=5a:94:ef:e4:0c:ee -device virtio-serial -chardev socket,path=/var/folders/mf/fk8hldv12_b3_j1vnp464th80000gn/T/podman/podman-machine-default_ready.sock,server=on,wait=off,id=apodman-machine-default_ready -device virtserialport,chardev=apodman-machine-default_ready,name=org.fedoraproject.port.0 -pidfile /var/folders/mf/fk8hldv12_b3_j1vnp464th80000gn/T/podman/podman-machine-default_vm.pid -machine q35,accel=hvf:tcg -cpu host -virtfs local,path=/Users/felder,mount_tag=vol0,security_model=mapped-xattr -drive if=virtio,file=/Users/felder/.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-36.20220918.2.2-qemu.x86_64.qcow2]
Waiting for VM ...
Error: qemu exited unexpectedly with exit code -1, stderr: qemu-system-x86_64: Error: HV_ERROR
```

**Describe the results you expected:**

```
$ /opt/podman/bin/podman machine start 
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users/felder:/Users/felder

This machine is currently configured in rootless mode. If your containers
require root permissions (e.g. ports < 1024), or if you run into compatibility
issues with non-podman clients, you can switch using the following command:
...
```

**Additional information you deem important (e.g. issue happens only occasionally):**

This issue is similar to #15834 but instead of having the bleeding edge macOS version this is for an older release. How many macOS releases are supported and are you building on older releases, e.g. Catalina?

**Output of `podman version`:**

There is no reasonable output b/c podman machine is not running.

This is version: **v4.2.1 (amd64)**

**Output of `podman info`:**

There is no reasonable output b/c podman machine is not running.

**Package info (e.g. output of `rpm -q podman` or `apt list podman` or `brew info podman`):**

No system package - macOS pkg file installation referenced above

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes (currently v4.2.1 is the latest release)

**Additional environment details (AWS, VirtualBox, physical, etc.):**

physical, MB Pro 2020, Intel Core i7

**Comment by baude** [comment]
@ashley-cui would you have time to peek at this? if not, maybe i can?

**Comment by ashley-cui** [comment]
@baude Looks like its an older OS, and I'm not sure how to get an older OS on a mac.

**Comment by rhatdan** [comment]
I don't think we can support older OSs

**Comment by baude** [comment]
correct, we do not; but this brings up a point about whether we documented what is supported as well as does the installer catch it ?

**Comment by cfelder** [comment]
Hi,

I suspect the qemu build on a more recent macOS version is not compatible w/ macOS Catalina (10.15.7)

I've resolved this issue by compiling [containers/podman-machine-qemu](github.com/containers/podman-machine-qemu) to ``/opt/podman/qemu`` on my own.

My build does not contain ``gvproxy`` in ``/opt/podman/qemu/bin`` but this is already shipped w/ the podman release package in ``/opt/podman/bin``. Thus I had to add ``helper_binaries_dir`` to my ``containers.conf`` file (see below). Otherwise ``gvproxy`` is not found on ``podman machine start``.

```toml
[engine]

helper_binaries_dir = ["/opt/podman/bin"]
```

It would be nice if the installer would use podman-machine-qemu build depending on the underlying macOS version. If that is not possible the installer should at least warn that this version in particular the qemu build may be not compatible w/ this macOS version.

I hope this sheds more light on this. If there is anything I can do to address this issue please let me know.

**Comment by rhatdan** [comment]
I don't believe this is something Podman can fix, but I am sure updated qemu will fix it.

**Comment by cfelder** [comment]
Indeed this must be fixed on ``podman-machine-qemu`` and is addressed in PR containers/podman-machine-qemu#8.
Nevertheless the installer ``pkg`` files are shipped in this repo.
